lib/floe/workflow/runner/kubernetes.rb

```diff
@@ -66,7 +68,17 @@ def status!(runner_context)
         end
 
         def running?(runner_context)
-          %w[Pending Running].include?(runner_context.dig("container_state", "phase"))
+          phase              = runner_context.dig("container_state", "phase")
+          container_statuses = runner_context.dig("container_state", "containerStatuses") || []
+          container_reasons  = container_statuses.map { |status| status.dig("state", "waiting", "reason") }.compact
+
+          return false unless RUNNING_PHASES.include?(phase)
+          # If an image failed to pull then the phase will be Pending but the
+          # container status will be waiting with a reason of ImagePullBackOff
+          # or ErrImagePull
+          return false unless (container_reasons & FAILURE_REASONS).empty?
```

Comment:
So it is not running, that means we will fetch a new status.
What `Error` will get assigned to the Task?

Reply:
This will return `false` if container_reasons includes any of the error failure states, and `success?` will return `false` since the phase isn't `Succeeded`.

Good point about the Task Error I should return the failure reason from `#output` if it is in this state.

Reply:
I think the `return false unless [].empty?` is needlessly difficult to grock and `return false if failures.any?` would read easier

Reply:
more_core_extensions has this as defined as `include_any?` (or `include_none?` if you want the opposite) for readbaility.

Reply:
Yeah trying to avoid more_core if possible, we only use it for one thing right now and we could refactor that and drop the dependency on that and activesupport all together.